src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,124 @@
+
+        assertEquals(registrationService.register(userToRegisterTwice), userToRegisterTwice);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userToRegisterTwice);
+        });
+    }
+
+    @Test
+    void register_userWithTooShortLogin_notOk() {
+        User userWithZeroSymbolsLogin = new User("", okPassword, okAge);
+        User userWithFiveSymbolsLogin = new User("johny", okPassword, okAge);
+        User userWithNullLogin = new User(null, okPassword, okAge);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithZeroSymbolsLogin);
+        });
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithFiveSymbolsLogin);
+        });
+
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(userWithNullLogin);
+        });
+    }
+
+    @Test
+    void register_userWithTooShortPassword_notOk() {
+        User userWithZeroSymbolsPassword = new User(okLoginOne, "", okAge);
+        User userWithFiveSymbolsPassword = new User(okLoginTwo, "qwert", okAge);
+        User userWithNullPassword = new User(okLoginThree, null, okAge);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithZeroSymbolsPassword);
+        });
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithFiveSymbolsPassword);
+        });
+
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_userWithTooYoungUser_notOk() {
+        User userWithZeroAge = new User(okLoginOne, okPassword, 0);
+        User userWithNegativeAge = new User(okLoginTwo, okPassword, -100);
+        User userWithEdgeCaseBadAge = new User(okLoginThree, okPassword, 17);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithZeroAge);
+        });
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNegativeAge);
+        });
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEdgeCaseBadAge);
+        });
+    }
+
+    @Test
+    void register_userWithRightParams_ok() {
+        User userToRegisterOne = new User(okLoginOne, okPassword, okAge);
+        User userToRegisterTwo = new User(okLoginTwo, okPassword, okAge);
+        User userToRegisterThree = new User(okLoginThree, okPassword, okAge);
+
+        assertEquals(registrationService.register(userToRegisterOne), userToRegisterOne);
+        assertEquals(registrationService.register(userToRegisterTwo), userToRegisterTwo);
+        assertEquals(registrationService.register(userToRegisterThree), userToRegisterThree);
```

Comment:
one ok case is enough
```suggestion
    void register_userWithRightParams_ok() {
        User userToRegisterOne = new User(okLoginOne, okPassword, okAge);

        assertEquals(registrationService.register(userToRegisterOne), userToRegisterOne);
```